test/core/iomgr/tcp_server_posix_test.cc

```diff
@@ -343,34 +341,34 @@ static void test_connect(size_t num_connects,
   resolved_addr.len = static_cast<socklen_t>(sizeof(struct sockaddr_storage));
   resolved_addr1.len = static_cast<socklen_t>(sizeof(struct sockaddr_storage));
   addr->ss_family = addr1->ss_family = AF_INET;
-  GPR_ASSERT(GRPC_LOG_IF_ERROR(
+  ASSERT_TRUE(GRPC_LOG_IF_ERROR(
       "grpc_tcp_server_add_port",
       grpc_tcp_server_add_port(s, &resolved_addr, &svr_port)));
   gpr_log(GPR_INFO, "Allocated port %d", svr_port);
-  GPR_ASSERT(svr_port > 0);
+  ASSERT_GT(svr_port, 0);
   /* Cannot use wildcard (port==0), because add_port() will try to reuse the
      same port as a previous add_port(). */
   svr1_port = grpc_pick_unused_port_or_die();
-  GPR_ASSERT(svr1_port > 0);
+  ASSERT_GT(svr1_port, 0);
   gpr_log(GPR_INFO, "Picked unused port %d", svr1_port);
   grpc_sockaddr_set_port(&resolved_addr1, svr1_port);
-  GPR_ASSERT(grpc_tcp_server_add_port(s, &resolved_addr1, &port) ==
-                 GRPC_ERROR_NONE &&
-             port == svr1_port);
+  ASSERT_TRUE(grpc_tcp_server_add_port(s, &resolved_addr1, &port) ==
```

Comment:
Debugging failures will be easier if you break this into two `ASSERT_EQ` statements.